src/multisig/sign/api.rs

```diff
@@ -0,0 +1,116 @@
+use super::r1;
+use crate::{
+    collections::{HoleVecMap, Subset, TypedUsize, VecMap},
+    crypto_tools::k256_serde,
+    gg20::sign::MessageDigest,
+    multisig::keygen::{
+        GroupPublicInfo, KeygenPartyId, KeygenShareId, SecretKeyShare, ShareSecretInfo,
+    },
+    sdk::{
+        api::{PartyShareCounts, Protocol, TofnFatal, TofnResult},
+        implementer_api::{new_protocol, ProtocolBuilder},
+    },
+};
+
+use serde::{Deserialize, Serialize};
+use tracing::error;
+
+/// SignProtocol output for a single share in happy path
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SignatureShare {
+    pub signature: k256_serde::Signature,
+    pub party_id: TypedUsize<KeygenPartyId>,
+    pub subshare_id: usize,
+}
+
+/// Exactly threshold + 1 valid signatures
+pub type SignProtocolOutput = Vec<SignatureShare>;
+
+/// Maximum byte length of messages exchanged during sign.
+/// The sender of a message larger than this maximum will be accused as a faulter.
+/// View all message sizes in the logs of the integration test `single_thred::basic_correctness`.
+/// The largest sign message is r2::P2pHappy with size ~6828 bytes on the wire.
+pub const MAX_MSG_LEN: usize = 7500;
```

Comment:
This should be updated. Same for keygen if it hasn't.

Reply:
Done.  I set 100 bytes as a conservative over-estimate to allow for cruft over the wire.  (Intermediate messages for both keygen/sign are ~32 bytes.)

Note that I can't easily get an exact length until we have integration tests for multisig.  Such tests would be a copy-paste of our integration tests for gg20.  That suggests to me that we might benefit from some abstractification for generic keygen/sign protocols.  Perhaps tofnd would also benefit from such abstractification @sdaveas .